proposals/improved-interpolated-strings.md

```diff
@@ -35,16 +35,16 @@ convenient interpolation syntax.
 We introduce a new builder pattern that can represent an interpolated string passed as an argument to a method. The simple English of the pattern is as follows:
 
 When an _interpolated\_string\_expression_ is passed as an argument to a method, we look at the type of the parameter. If the parameter type has a static method
-`GetInterpolatedStringBuilder` that can invoked with 2 int parameters, `baseLength` and `formatHoleCount`, optionally takes a parameter the receiver is convertible to,
-and has an out parameter of the type of original method's parameter and that type has instance `TryFormat` methods can be invoked for every part of the interpolated
+`Create` that can invoked with 2 int parameters, `baseLength` and `formatHoleCount`, optionally takes a parameter the receiver is convertible to,
+and has an out parameter of the type of original method's parameter and that type has instance `TryFormatX` methods can be invoked for every part of the interpolated
 string, then we lower the interpolation using that, instead of into a traditional call to `string.Format(formatStr, args)`. A more concrete example is helpful for
 picturing this:
 
 ```cs
 // The builder that will actually "build" the interpolated string"
 public ref struct TraceLoggerParamsBuilder
 {
-    public static bool GetInterpolatedStringBuilder(int baseLength, int formatHoleCount, Logger logger, out TraceLoggerParamsBuilder builder)
+    public static bool Create(int baseLength, int formatHoleCount, Logger logger, out TraceLoggerParamsBuilder builder)
```

Comment:
One of the suggestions from the previous PR was to make these methods return the builder type and have the `bool` as the `out` parameter. Is there a reason we didn't change to use that form? 

The current form is less efficient performance wise (returning complex types is faster than initializing them by `ref / out`) and also has implications with `ref struct` life time rules. A non-trivial amount of our interpolated builders are going to be `ref struct` and once they are passed as  `ref / out` then we need to do a lot of due dilligence around the life time rules here. 

The details here are mundane but important and the `ref` field proposal probably does the best job of explaining them 

https://github.com/dotnet/csharplang/blob/main/proposals/low-level-struct-improvements.md#provide-parameter-does-not-escape-annotations

In short though we would need to make guarantees about the lifetimes of the temporaries that we pass as the `out` here. No such guarantees need to be done if we simple return them here.

Reply:
Discussed this offline. While LDM hasn't approved that change currently, I'll go ahead and make the change, and leave an open question to confirm it.